src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,162 @@
+import _ from 'underscore';
+import React from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import moment from 'moment';
+import Text from '../Text';
+import ArrowIcon from './ArrowIcon';
+import styles from '../../styles/styles';
+import {propTypes as calendarPickerPropType, defaultProps as defaultCalendarPickerPropType} from './calendarPickerPropTypes';
+import generateMonthMatrix from './generateMonthMatrix';
+import withLocalize from '../withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+
+class CalendarPicker extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.monthNames = moment.localeData(props.preferredLocale).months();
+        this.daysOfWeek = moment.localeData(props.preferredLocale).weekdays();
+
+        let currentDateView = props.value;
+        if (props.selectedYear) {
+            currentDateView = moment(currentDateView).set('year', props.selectedYear).toDate();
+        }
+        if (props.maxDate < currentDateView) {
+            currentDateView = props.maxDate;
+        } else if (props.minDate > currentDateView) {
+            currentDateView = props.minDate;
+        }
+
+        this.state = {
+            currentDateView,
+        };
+
+        this.moveToPrevMonth = this.moveToPrevMonth.bind(this);
+        this.moveToNextMonth = this.moveToNextMonth.bind(this);
+        this.onYearPickerPressed = this.onYearPickerPressed.bind(this);
+        this.onDayPressed = this.onDayPressed.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.minDate <= this.props.maxDate) {
+            return;
+        }
+        throw new Error('Minimum date cannot be greater than the maximum date.');
+    }
+
+    componentDidUpdate(prevProps) {
+        // Check if selectedYear has changed
+        if (this.props.selectedYear === prevProps.selectedYear) {
+            return;
+        }
+
+        // If only the selectedYear prop has changed, update the currentDateView state with the new year value
+        this.setState(prev => ({currentDateView: moment(prev.currentDateView).set('year', this.props.selectedYear).toDate()}));
+    }
+
+    /**
+     * Handles the user pressing the year picker button.
+     * Opens the year selection screen with the minimum and maximum year range
+     * based on the props, the current year based on the state, and the active route.
+     */
+    onYearPickerPressed() {
+        const minYear = moment(this.props.minDate).year();
+        const maxYear = moment(this.props.maxDate).year();
+        const currentYear = this.state.currentDateView.getFullYear();
+        Navigation.navigate(ROUTES.getYearSelectionRoute(minYear, maxYear, currentYear, Navigation.getActiveRoute()));
+    }
+
+    /**
+     * Calls the onSelected function with the selected date, if it is within the min/max range.
```

Comment:
I don't see the check for `if it is within the min/max range.` in this function, am I confused?

Reply:
@Beamanator Ah nice catch, that logic was removed it was redundant. Days that do not fall in the min/max range can't be pressed in the first place.